proxy/src/logging.rs

```diff
@@ -94,3 +129,878 @@ impl Drop for LoggingGuard {
+                    "fields",
+                    &SerializableEventFields(event, skipped_field_indices),
+                )?;
+            }
+
+            let pid = std::process::id();
+            if pid != 1 {
+                serializer.serialize_entry("process_id", &pid)?;
+            }
+
+            serializer.serialize_entry("thread_id", &THREAD_ID.get())?;
+
+            // TODO: tls cache? name could change
+            if let Some(thread_name) = std::thread::current().name() {
+                if !thread_name.is_empty() && thread_name != "tokio-runtime-worker" {
+                    serializer.serialize_entry("thread_name", thread_name)?;
+                }
+            }
+
+            if let Some(task_id) = tokio::task::try_id() {
+                serializer.serialize_entry("task_id", &format_args!("{task_id}"))?;
+            }
+
+            serializer.serialize_entry("target", meta.target())?;
+
+            if let Some(module) = meta.module_path() {
+                if module != meta.target() {
+                    serializer.serialize_entry("module", module)?;
+                }
+            }
+
+            if let Some(file) = meta.file() {
+                if let Some(line) = meta.line() {
+                    serializer.serialize_entry("src", &format_args!("{file}:{line}"))?;
+                } else {
+                    serializer.serialize_entry("src", file)?;
+                }
+            }
+
+            {
+                let otel_context = Span::current().context();
+                let otel_spanref = otel_context.span();
+                let span_context = otel_spanref.span_context();
+                if span_context.is_valid() {
+                    serializer.serialize_entry(
+                        "trace_id",
+                        &format_args!("{}", span_context.trace_id()),
+                    )?;
+                }
+            }
+
+            serializer.serialize_entry("spans", &SerializableSpanStack(ctx))?;
+
+            serializer.end()
+        };
+
+        serialize().map_err(io::Error::other)?;
+        self.logline_buffer.push(b'\n');
+        Ok(())
+    }
+}
+
+/// Serializes the tracing lavel.
+struct SerializeLevel<'a>(&'a Level);
+
+impl Serialize for SerializeLevel<'_> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        // This order was chosen to match log level probabilities in prod.
+        if self.0 == &Level::INFO {
+            serializer.serialize_str("INFO")
+        } else if self.0 == &Level::WARN {
+            serializer.serialize_str("WARN")
+        } else if self.0 == &Level::ERROR {
+            serializer.serialize_str("ERROR")
+        } else if self.0 == &Level::DEBUG {
+            serializer.serialize_str("DEBUG")
+        } else if self.0 == &Level::TRACE {
+            serializer.serialize_str("TRACE")
+        } else {
+            unreachable!("unknown log level: {}", self.0)
+        }
```

Comment:
```suggestion
        serializer.serialize_str(self.0.as_str())
```
This seems to produce better assembly via a switch table

Reply:
Done.

In the previous incarnation of the subscriber I was experimenting with different and shorter and also equally long names for the levels, but went back to identical names here and didn't realize I can go to just using as_str.